### PR TITLE
Add support for merge statements

### PR DESCRIPTION
**Pull request by vchan**

### CODE REVIEW

File: sqlglot/parser.py [hunk 1]
```diff
@@ -408,6 +408,7 @@ class Parser(metaclass=_Parser):
         TokenType.COMMIT: lambda self: self._parse_commit_or_rollback(),
         TokenType.END: lambda self: self._parse_commit_or_rollback(),
         TokenType.ROLLBACK: lambda self: self._parse_commit_or_rollback(),
+        TokenType.MERGE: lambda self: self._parse_merge(),
```
Comment (tobymao) [inline sqlglot/parser.py:411]: add this to id var tokens

**Comment by tobymao** [comment]
looks good

**Review by GeorgeSittas — APPROVED** [review]

**Comment by GeorgeSittas** [comment]
Agree with Toby, seems like only some minor syntax pieces are missing.

File: sqlglot/parser.py [hunk 2]
```diff
@@ -2650,6 +2653,46 @@ def _parse_set_item(self):
...
+                    expression=self._match(TokenType.VALUES) and self._parse_value(),
+                )
+            elif self._match(TokenType.UPDATE):
+                then = self.expression(
+                    exp.Update,
+                    expressions=self._match(TokenType.SET)
+                    and self._parse_csv(self._parse_equality),
+                )
+            elif self._match(TokenType.DELETE):
+                then = self.expression(exp.Var, this=TokenType.DELETE.name)
```
Comment (GeorgeSittas) [inline sqlglot/parser.py:2684]:
```suggestion
            elif self._match(TokenType.DELETE):
                then = self.expression(exp.Var, this=self._prev.text)
```

**Pull request merged**